drivers/rtc/Kconfig.counter

```diff
@@ -0,0 +1,19 @@
+# Copyright 2025 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config RTC_COUNTER
+	bool "Real Time Clock driver using counter API"
+	default y
+	depends on DT_HAS_ZEPHYR_COUNTER_RTC_ENABLED
+	help
+	  RTC driver applied for RTC device without broken-down time registers.
+	  Wraps the counter API to provide RTC functionality.
+	  Uses unix timestamp to convert between broken-down time and counter ticks.
+
+config RTC_COUNTER_MAX_ALARMS
+	int "Maximum number of RTC alarm channels exposed by counter"
+	default 1
+	depends on RTC_COUNTER && RTC_ALARM
+	help
+	  Caps how many counter alarm channels are exposed as RTC alarm IDs [0..N-1].
+	  If the underlying counter provides more channels, only the first N are used.
```

Comment:
This value is specific to the device, and is set with the dt propery `alarms-count = <n>`, the driver can get this prop to create the appropriate number of alarm contexts